source/magpie/group.d

```diff
@@ -699,24 +729,36 @@ public:
 
         static if(axis)
         {
-            static foreach(i; 0 .. GrpRowType.length)
-                if(i == pos[1])
-                {
-                    static if(kind == Universal)
-                        Slice!(Type*, 1, kind) ret = slice!(Type)(elementCountTill[pos[0] + 1] - elementCountTill[pos[0]]).universal;
-                    else static if(kind == Canonical)
-                        Slice!(Type*, 1, kind) ret = slice!(string)(elementCountTill[pos[0] + 1] - elementCountTill[pos[0]]).canonical;
-                    else
-                        Slice!(Type*, 1, kind) ret = slice!(string)(elementCountTill[pos[0] + 1] - elementCountTill[pos[0]]);
-
-                    static if(__traits(isArithmetic, Type, GrpRowType[i]) || is(Type == GrpRowType[i]))
-                        foreach(j; elementCountTill[pos[0]] .. elementCountTill[pos[0] + 1])
-                            ret[j - elementCountTill[pos[0]]] = cast(Type)data[i][j];
 
-                    return ret;
-                }
+            static if(kind == Universal)
+                Slice!(Type*, 1, kind) ret = slice!(Type)(elementCountTill[pos[0] + 1] - elementCountTill[pos[0]]).universal;
+            else static if(kind == Canonical)
+                Slice!(Type*, 1, kind) ret = slice!(string)(elementCountTill[pos[0] + 1] - elementCountTill[pos[0]]).canonical;
```

Comment:
this should either be `Slice!(string*, 1, kind) ret = slice!(string)` or Slice!(Type*, 1, kind) ret = slice!(Type)`